xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/PropertyClass.java

```diff
@@ -844,4 +852,23 @@ private String getTypeName()
     {
         currentProperty.merge(previousProperty, newProperty, configuration, context, mergeResult);
     }
+
+    /**
+     * Render the provided text to prevent XSS.
+     *
+     * @param text the text to escape.
+     * @param context the xwiki context.
+     * @return the text once sanitized.
+     */
```

Comment:
Missing `@Unstable`. I don't like the method name. You say it's about rendering and it's named sanitize. IMO the name should not do the rendering at all. 

Also I have the feeling the code is not in the right place. It seems like it should be in XWikiDocument since it calls getRenderedContent() on a XWikiDocument.

Why do you remove a paragraph? Is it always HTML?

Reply:
Where is the sanitization?